isisd/isis_cli.c

```diff
@@ -343,19 +339,49 @@ DEFPY(isis_bfd,
 		return CMD_SUCCESS;
 	}
 
-	nb_cli_enqueue_change(vty, "./frr-isisd:isis/bfd-monitoring",
+	nb_cli_enqueue_change(vty, "./frr-isisd:isis/bfd-monitoring/enabled",
 			      NB_OP_MODIFY, no ? "false" : "true");
 
 	return nb_cli_apply_changes(vty, NULL);
 }
 
+/*
+ * XPath: /frr-interface:lib/interface/frr-isisd:isis/bfd-monitoring/profile
+ */
+DEFPY(isis_bfd_profile, isis_bfd_profile_cmd, "[no] isis bfd profile WORD",
+      NO_STR PROTO_HELP
+      "Enable BFD support\n"
+      "Use a pre-configured profile\n"
+      "Profile name\n")
+{
+	const struct lyd_node *dnode;
+
+	dnode = yang_dnode_get(vty->candidate_config->dnode,
+			       "%s/frr-isisd:isis", VTY_CURR_XPATH);
+	if (dnode == NULL) {
+		vty_out(vty, "ISIS is not enabled on this circuit\n");
+		return CMD_SUCCESS;
+	}
+
+	nb_cli_enqueue_change(vty, "./frr-isisd:isis/bfd-monitoring/profile",
+			      NB_OP_MODIFY, no ? NULL : profile);
+
+	return nb_cli_apply_changes(vty, NULL);
+}
+
 void cli_show_ip_isis_bfd_monitoring(struct vty *vty, struct lyd_node *dnode,
 				     bool show_defaults)
 {
-	if (!yang_dnode_get_bool(dnode, NULL))
+	const char *profile;
+
+	if (!yang_dnode_get_bool(dnode, "./enabled"))
 		vty_out(vty, " no");
 
 	vty_out(vty, " isis bfd\n");
+
+	profile = yang_dnode_get_string(dnode, "./profile");
+	if (profile[0] != '\0')
+		vty_out(vty, " isis bfd profile %s\n", profile);
```

Comment:
Is it supposed to show `isis bfd` twice (e.g. one for `isis bfd` and another for `isis bfd profile <foo>`)?

Another thing is this might break if no profile is available... I think the proper check would be:

```c
if (yang_dnode_exists(dnode, "./profile"))
	vty_out(vty, " isis bfd profile %s\n", yang_dnode_get_string(dnode, "./profile");
```

Reply:
Yea, there are two different commands:
1. `isis bfd`
2. `isis bfd profile BFDPROF`
So you ahve two lines of config.